lib/parse.js

```diff
@@ -1,54 +1,21 @@
 "use strict";
 
 const babylonToEspree = require("./babylon-to-espree");
-const parse = require("@babel/parser").parse;
-const tt = require("@babel/parser").tokTypes;
-const traverse = require("@babel/traverse").default;
-const codeFrameColumns = require("@babel/code-frame").codeFrameColumns;
+const { parseSync: parse, tokTypes: tt, traverse } = require("@babel/core");
 
 module.exports = function(code, options) {
-  const legacyDecorators =
-    options.ecmaFeatures && options.ecmaFeatures.legacyDecorators;
-
   const opts = {
-    codeFrame: options.hasOwnProperty("codeFrame") ? options.codeFrame : true,
     sourceType: options.sourceType,
-    allowImportExportEverywhere: options.allowImportExportEverywhere, // consistent with espree
-    allowReturnOutsideFunction: true,
-    allowSuperOutsideMethod: true,
-    ranges: true,
-    tokens: true,
-    plugins: [
-      ["flow", { all: true }],
-      "jsx",
-      "estree",
-      "asyncFunctions",
-      "asyncGenerators",
-      "classConstructorCall",
-      "classProperties",
-      legacyDecorators
-        ? "decorators-legacy"
-        : ["decorators", { decoratorsBeforeExport: false }],
-      "doExpressions",
-      "exponentiationOperator",
-      "exportDefaultFrom",
-      "exportNamespaceFrom",
-      "functionBind",
-      "functionSent",
-      "objectRestSpread",
-      "trailingFunctionCommas",
-      "dynamicImport",
-      "numericSeparator",
-      "optionalChaining",
-      "importMeta",
-      "classPrivateProperties",
-      "bigInt",
-      "optionalCatchBinding",
-      "throwExpressions",
-      ["pipelineOperator", { proposal: "minimal" }],
-      "nullishCoalescingOperator",
-      "logicalAssignment",
-    ],
+    configFile: options.configFile,
+    filename: options.filePath,
```

Comment:
I think we should probably pass through all the options that can directly influence a config:

* `cwd`
* `root`
* `rootMode`
* `envName`
* `configFile`
* `babelrc`
* `babelrcRoots`
* `extends`
* `env`
* `overrides`
* `test`
* `include`
* `exclude`
* `ignore`
* `only`

It would be great to also set
```
caller: {
  name: "babel-eslint",
},
```

Reply:
I have updated this, but I'm wondering if we do really want the config merging options here - shouldn't we be able to just provide the config loading options that would point to the config file that would contain the config merging options? This would avoid duplication in the `.eslintrc.js` and `babel.config.js` files.

Reply:
I personally would prefer to start with fewer rather than more options (especially since this project is already difficult to maintain), but am happy to defer to what you think is best. :+1: